app/controllers/gigs.js

```diff
@@ -176,12 +176,45 @@ module.exports = function(app) {
           "Authorization": GROOT_ACCESS_TOKEN
         },
         json: true,
-        body: {}
+        body: {
+          active: false
+        }
       }, function(err, response, body) {
         return res.send(body);
       });
     });
   });
+  app.post('/intranet/gigs/edit/:gig_id', function(req, res) {
+    request({
+      url: `${SERVICES_URL}/gigs/${req.params.gig_id}`,
+      method: "GET",
+      headers: {
+        "Authorization": GROOT_ACCESS_TOKEN,
+      },
+      json: true
+    }, function(err, response, body) {
+      if(body.issuer != req.session.student.netid) {
```

Comment:
Yeah, I've thought that too. There is a lot of code duplication, and most of it never gets tested...